lawnchair/src/app/lawnchair/smartspace/IcuDateTextView.kt

```diff
@@ -39,27 +51,62 @@ class IcuDateTextView @JvmOverloads constructor(
 
     private fun onTimeChanged(updateFormatter: Boolean) {
         if (isShown) {
-            if (formatter == null || updateFormatter) {
-                val formatter: DateFormat = DateFormat.getInstanceForSkeleton(
-                    context.getString(R.string.smartspace_icu_date_pattern), Locale.getDefault()
-                )
-                this.formatter = formatter
-                formatter.setContext(DisplayContext.CAPITALIZATION_FOR_BEGINNING_OF_SENTENCE)
+            val format = when (calendar) {
+                SmartspaceCalendar.Persian -> getTimeTextPersian(updateFormatter = updateFormatter)
+                else -> getTimeTextGregorian(updateFormatter = updateFormatter)
             }
-            val format = formatter!!.format(java.lang.Long.valueOf(System.currentTimeMillis()))
             if (text != format) {
+                textAlignment =
+                    if (calendar == SmartspaceCalendar.Persian) TEXT_ALIGNMENT_TEXT_END else TEXT_ALIGNMENT_TEXT_START
                 text = format
                 contentDescription = format
             }
         }
     }
 
+    private fun getTimeTextPersian(updateFormatter: Boolean): String {
+        if (formatterPersian == null || updateFormatter) {
+            val formatter = PersianDateFormat(
+                context.getString(R.string.smartspace_icu_date_pattern_persian),
+                PersianDateFormat.PersianDateNumberCharacter.FARSI
+            )
+            this.formatterPersian = formatter
+        }
+        return formatterPersian!!.format(PersianDate(java.lang.Long.valueOf(System.currentTimeMillis())))
+    }
+
+    private fun getTimeTextGregorian(updateFormatter: Boolean): String {
+        if (formatterGregorian == null || updateFormatter) {
+            val formatter: DateFormat = DateFormat.getInstanceForSkeleton(
+                context.getString(R.string.smartspace_icu_date_pattern_gregorian),
+                Locale.getDefault()
+            )
+            this.formatterGregorian = formatter
+            formatter.setContext(DisplayContext.CAPITALIZATION_FOR_BEGINNING_OF_SENTENCE)
+        }
+        return formatterGregorian!!.format(java.lang.Long.valueOf(System.currentTimeMillis()))
```

Comment:
I’d suggest avoiding the use of not-null assertion here.

```suggestion
        val formatter = formatterGregorian.takeIf { updateFormatter.not() }
            ?: DateFormat.getInstanceForSkeleton(
                context.getString(R.string.smartspace_icu_date_pattern_gregorian),
                Locale.getDefault()
            ).also { formatterGregorian = it }
        return formatter.format(java.lang.Long.valueOf(System.currentTimeMillis()))
```

Reply:
This is the exact original code and I just tried to not modify it and keep the same format even for the Persian Date function. It's now fixed on the new force push.

Reply:
Ah, sorry, I didn’t notice that you’d used the preexisting code. If I had, I wouldn’t have requested this change, as it’s not entirely related to the pull request. I would’ve probably made it myself after the merge. But if you’ve already applied it, I don’t think it hurts to keep it. My apologies for the error.